### PR TITLE
minimal hooks for SN32 flashing support

### PR DESCRIPTION
**Pull request by dexter93**
Bare essentials for `qmk flash` support in sn32 boards.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by sigprof — CHANGES_REQUESTED** [review]

**Comment by dexter93** [comment]
cli integration status:

- macOS: https://github.com/qmk/homebrew-qmk/pull/63
- windows: https://github.com/msys2/MINGW-packages/pull/17529
- linux: direct from https://github.com/SonixQMK/SonixFlasherC/releases

**Comment by dexter93** [comment]
PR is good to go. Review at your convenience

**Comment by dexter93** [comment]
with #21217 now merged, bumping this in hopes to make it in this merge cycle.

File: lib/python/qmk/cli/doctor/check.py [hunk 1]
```diff
@@ -44,7 +44,12 @@ def _check_arm_gcc_version():
     version_number = ESSENTIAL_BINARIES['arm-none-eabi-gcc']['output'].strip()
     cli.log.info('Found arm-none-eabi-gcc version %s', version_number)
 
-    return CheckStatus.OK  # Right now all known arm versions are ok
+    parsed_version = _parse_gcc_version(version_number)
+    if (parsed_version['major'], parsed_version['minor']) < (10, 3):
+        cli.log.warning('{fg_yellow}We do not recommend arm-none-eabi-gcc older than 10.3.x. Upgrading to 10.3.x or higher is recommended.')
+        return CheckStatus.WARNING
+
+    return CheckStatus.OK  # arm versions less than 10.3.x cause issues on sn32
```
Comment (fauxpark) [inline lib/python/qmk/cli/doctor/check.py:52]: What is the reason for this?

Comment (dexter93) [reply lib/python/qmk/cli/doctor/check.py:52]: the reason for this is occasional and random hardfaults when built against those versions in arm m0. The toolchain version that first fixed it was `arm-none-eabi-gcc (GNU Arm Embedded Toolchain 10-2020-q4-major)`. Bumping the version one up makes sure the right one is selected ( up until q2 it was buggy). No known issues exist with current latest version, tested extensively with every major release from 10.3 onwards.

while my original notes are lost on this, I believe this to be the [bug](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=91816)

**Comment by tzarc** [comment]
After internal discussion, this isn't in alignment with the strategic direction QMK has chosen to take. Thanks for the effort put in, but we've chosen not to include this PR with QMK.

**Pull request closed without merging**